raiwidgets/raiwidgets/_cohort.py

```diff
@@ -140,39 +183,286 @@ def _validate_cohort_filter_parameters(
     def add_cohort_filter(self, cohort_filter: CohortFilter):
         """Add a cohort filter into the cohort.
-
         :param cohort_filter: Cohort filter defined by CohortFilter class.
         :type: CohortFilter
         """
+        if not isinstance(cohort_filter, CohortFilter):
+            raise UserConfigValidationException(
+                "Got unexpected type {0} for cohort filter. "
+                "Expected CohortFilter type.".format(type(cohort_filter))
+            )
         if self.cohort_filter_list is None:
             self.cohort_filter_list = [cohort_filter]
         else:
             self.cohort_filter_list.append(cohort_filter)
+
+    def _validate_with_test_data(self, test_data: pd.DataFrame,
+                                 target_column: str,
+                                 categorical_features: List[str],
+                                 is_classification: Optional[bool] = True):
+        """
+        Validate the cohort and cohort filters parameters with respect to
+        test data.
+
+        :param test_data: Test data over which cohort analysis will be done
+            in ResponsibleAI Dashboard.
+        :type test_data: pd.DataFrame
+        :param target_column: The target column in the test data.
+        :type target_column: str
+        :param categorical_features: The categorical feature names.
+        :type categorical_features: list[str]
+        :param is_classification: True to indicate if this validation needs
+            to be done for a classification scenario and False to indicate
+            that this needs to be done for regression scenario.
+        :type is_classification: bool
+        """
+        if self.cohort_filter_list is None:
+            return
+        if not isinstance(test_data, pd.DataFrame):
+            raise UserConfigValidationException(
+                "The test_data should be a pandas DataFrame.")
+        if not isinstance(target_column, str):
+            raise UserConfigValidationException(
+                "The target_column should be string.")
+        if not isinstance(categorical_features, list):
+            raise UserConfigValidationException(
+                "Expected a list type for categorical columns.")
+        if not all(isinstance(entry, str) for entry in categorical_features):
+            raise UserConfigValidationException(
+                "All entries in categorical_features need of string type."
+            )
+
+        if target_column not in test_data.columns:
+            raise UserConfigValidationException(
+                "The target_column {0} was not found in test_data.".format(
+                    target_column)
+            )
+
+        test_data_columns_set = set(test_data.columns) - set([target_column])
+        if not all(entry in test_data_columns_set
+                   for entry in categorical_features):
+            raise UserConfigValidationException(
+                "Found some categorical feature name which is not"
```

Comment:
maybe specify via categorucal_features parameter?  so user knows where "categorical feature name" is coming from?

Reply:
Done. Added column name to exception to guide the user.